libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -868,29 +1059,90 @@ def batch(
 
         # invoke
         try:
-            for step in self.steps:
-                inputs = step.batch(
-                    inputs,
-                    [
-                        # each step a child run of the corresponding root run
-                        patch_config(config, callbacks=rm.get_child())
-                        for rm, config in zip(run_managers, configs)
-                    ],
-                )
+            if return_exceptions:
```

Comment:
getting hard to read, should we factor out some logic into helper functions?

Reply:
This would be a helper function that cannot be reused, as this logic is specific RunnableSequence.batch()

Reply:
sure but think there's some argument to be made for factoring out just for readability? up to you